core/opendaq/opendaq/tests/test_instance.cpp

```diff
@@ -307,5 +307,109 @@ TEST_F(InstanceTest, Serialize)
     std::cout << str << std::endl;
 }
 
+TEST_F(InstanceTest, InstanceBuilderSetGet)
```

Comment:
I'd just add a test that validates that the correct the correct fields are ignored when setting incompatible options. Eg. that the "Scheduler worker num" is ignored when the Scheduler is set.